src/bootstrap/test.rs

```diff
@@ -965,6 +965,10 @@ impl Step for Compiletest {
             cmd.arg("--bless");
         }
 
+        if let Some(compare_mode) = builder.config.cmd.compare_mode() {
+            cmd.arg("--compare-mode").arg(compare_mode);
+        }
+
```

Comment:
There appears to be some very similar logic a few lines below:

https://github.com/spastorino/rust/blob/fb2a51225f26f8acf39e482d916f112265c81d35/src/bootstrap/test.rs#L1136-L1143

It seems like there is a `compare_mode` that comes from the test directory configuration as well as one (now) that you have added to `builder.config.cmd`. Perhaps we want to just change [this line](https://github.com/spastorino/rust/blob/fb2a51225f26f8acf39e482d916f112265c81d35/src/bootstrap/test.rs#L890) to something like this:

```rust
let compare_mode = builder.config.cmd.compare_mode().or(self.compare_mode);
```

(This is giving precedence to the `--compare-mode` option from the command line, which seems appropriate.)

(Alternatively, we could error if compare-mode is given *both* in the command line *and* the test suite definition; I think that the latter (test suite) can only occur when running `./x.py test`, in which case the command line probably ought not to be in use.)